preview-env/create/action.yml

```diff
@@ -133,3 +133,19 @@ runs:
       env_url: ${{ inputs.app_url }}
       ref: ${{ inputs.revision }}
       env: ${{ inputs.app_name }}
+  - name: Create deployment status artifact
+    if: always()
+    shell: bash
+    run: |
+      if [[ "${{ github.action_status }}" == "success" ]]; then
+        echo ":full_moon_with_face: [\`${{ inputs.app_name }}\`](${{ inputs.app_url }}): ${{ github.action_status }}\n" > status_${{ inputs.app_name }}.md
+      else
+        echo ":boom: \`${{ inputs.app_name }}\`: ${{ github.action_status }} :arrow_right: see [ArgoCD](https://${{ inputs.argocd_server }}/applications?health=Progressing%2CDegraded&search=${{ inputs.app_name }})\n" > status_${{ inputs.app_name }}.md
+      fi
```

Comment:
I think you can make this step more readable by defining the two different templates in separate env variables at the action level, and use YAML multi-line capabilities ... 
Same for the `github.action_status` value, as from a security point of view it is recommended to avoid injecting context values directly into the script, and to use environment variables instead.

```
  - name: Create deployment status artifact
    if: always()
    shell: bash
    env:
      TEMPLATE_A: >
        my
        template A
      TEMPLATE_B: |
        my template B
      STATUS: ${{ github.action_status }}
    run: |
      if [[ "$STATUS" == "success" ]]; then
        echo "$TEMPLATE_A" > status_${{ inputs.app_name }}.md
      else
        echo "$TEMPLATE_B" > status_${{ inputs.app_name }}.md
      fi
```